### PR TITLE
Add example quines and ouroboros functions

### PR DESCRIPTION
**Pull request by michaeladda**
Manually tested (with the included test script)
Please note linting is deliberately not done as it impacts the code.

### CODE REVIEW

File: quines/README.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# Binaris Quine Collection
```
Comment (rylandg) [inline quines/README.md]: Binaris might not be needed considering you're in our function examples repo.

File: quines/README.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+A [quine](https://en.wikipedia.org/wiki/Quine_(computing)) is a non-empty computer program which takes no input and produces a copy of its own source code as its only output.  
+An [ouroboros](https://en.wikipedia.org/wiki/Ouroboros) is a serpent eating its own tail, and a recursion of the quine concept.
```
Comment (rylandg) [inline quines/README.md]: Not sure I like the wording of 

> and a recursion of the quine concept.

File: quines/README.md [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+
+You'll find here two quines, one in JavaScript, and one in Python, and "one" ouroboros in JavaScript and Python.
```
Comment (rylandg) [inline quines/README.md]: Maybe

"This repo contains two quines, one in JavaScript, and one in Python. It also contains  "one" ouroboros in JavaScript and Python."

Note that the subtle "one" joke/comment will probably be lost on readers who don't come in with entry knowledge.

File: quines/README.md [hunk 4]
```diff
@@ -0,0 +1,11 @@
+# Binaris Quine Collection
+
+A [quine](https://en.wikipedia.org/wiki/Quine_(computing)) is a non-empty computer program which takes no input and produces a copy of its own source code as its only output.  
+An [ouroboros](https://en.wikipedia.org/wiki/Ouroboros) is a serpent eating its own tail, and a recursion of the quine concept.
+
+You'll find here two quines, one in JavaScript, and one in Python, and "one" ouroboros in JavaScript and Python.  
+The quine functions will print themselves when invoked, and the ouroboros functions will print each other when invoked.
```
Comment (rylandg) [inline quines/README.md]: Suggest
"The quine functions print themselves when invoked, and the ouroboros functions print each other when invoked."

File: quines/README.md [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+
+You can deploy the functions with `./deploy.sh`
```
Comment (rylandg) [inline quines/README.md]: Are these meant to be steps?

File: quines/ouroboros.js [hunk 6]
```diff
@@ -0,0 +1,27 @@
+const js = `
+const quote = String.fromCharCode(34);
+const newline = String.fromCharCode(10);
+const marker = quote + quote + quote;
+const quine = 'js = ' + marker + js + marker + newline + 'py = ' + marker + py + marker + py;
+exports.handler = async (body, ctx) => {
+  return new ctx.Response({
```
Comment (rylandg) [inline quines/ouroboros.js]: You use `Response` here but not  elsewhere (line 17), I know they are different languages but it's probably really weird to see as an outside viewer.

File: quines/ouroboros.js [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+def handler(body, ctx):
+    return ctx.HTTPResponse(body='{}'.format(quine))
+`;
+const quote = String.fromCharCode(34);
+const newline = String.fromCharCode(10);
+const marker = quote + quote + quote;
+const quine = 'js = ' + marker + js + marker + newline + 'py = ' + marker + py + marker + py;
+exports.handler = async (body, ctx) => {
+  return new ctx.Response({
+    body: Buffer.from(quine),
```
Comment (rylandg) [inline quines/ouroboros.js]: Not sure I understand the need for Buffer.

File: quines/README.md (hunk 2 above)
Comment (michaeladda) [reply quines/README.md]: it's based on https://en.wikipedia.org/wiki/Quine_(computing)#Ouroboros_programs
you're the native english speaker. suggest something better :-)

File: quines/README.md (hunk 5 above)
Comment (michaeladda) [reply quines/README.md]: these are hardly steps. just a direction to the helper scripts.

File: quines/ouroboros.js (hunk 7 above)
Comment (michaeladda) [reply quines/ouroboros.js]: remnants from obsolete interface. removed.

File: quines/ouroboros.js (hunk 6 above)
Comment (michaeladda) [reply quines/ouroboros.js]: Response is the obsolete interface. upgraded and now matching the python syntax.

File: quines/README.md (hunk 2 above)
Comment (rylandg) [reply quines/README.md]: 1. Even if Wiki is correct most of the time it doesn't necessarily mean it's digestable most of the time.
2. The first thing I did was copy your sentence and then visit the Wiki using 'c-f'. "recurs" is only found twice and definitely not with your wording.

My suggestion would be.

"is a serpent eating its own tail, and can be thought of as a 'recursive' quine."

File: quines/README.md (hunk 5 above)
Comment (rylandg) [reply quines/README.md]: It's just a bit awkward because you start a sentence with "Now,"

**Comment by michaeladda** [comment]
@rylandg Thanks! Further improved the readme. if it can still be improved, please don't hesitate to open an issue and assign it to me :-)

**Pull request merged**